### ISSUE TITLE
Declare `InlayHint` colours

### ISSUE THREAD

**Issue by cyqsimon** [akamud/vscode-theme-onedark#146]
VSCode now enables themes to define [colours for inline hints (parameter & type)](https://code.visualstudio.com/api/references/theme-color#:~:text=editorInlayHint.background). Unfortunately the default values don't fit One Dark particularly well:

![image](https://user-images.githubusercontent.com/28627918/162987511-c1ca4b42-378f-4366-9ff4-68124f9b3d78.png)

They are hints and should be much subtler than that.

So I took a look at [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme) and based my values off theirs (see [here](https://github.com/Binaryify/OneDark-Pro/blob/97363c147989d77360f638112fe3cb14f7c11e15/src/themes/data/oneDarkPro.ts#L22-L23)). I lowered the contrast a bit to my liking.

```json
"editorInlayHint.background": "#2c323d",
"editorInlayHint.foreground": "#9ba1ad",
```

Declaring them makes it look much nicer:

![image](https://user-images.githubusercontent.com/28627918/162992903-eee3f39e-9964-4f6c-89af-5488658b5a59.png)

## Sample code

Taken from [sharkdp/bat](https://github.com/sharkdp/bat/blob/520081a92c73652653c4f2b40e9527dd36b234bb/src/bin/bat/main.rs#L236-L268):

You will need to install the [Rust toolchain](https://www.rust-lang.org/tools/install) and [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=matklad.rust-analyzer) to see the hints though, so this may not be the best example. Any supported language with type inference should do.

```rust
    let mut report = bugreport!()
        .info(SoftwareVersion::default())
        .info(OperatingSystem::default())
        .info(CommandLine::default())
        .info(EnvironmentVariables::list(&[
            "SHELL",
            "PAGER",
            "LESS",
            "LANG",
            "LC_ALL",
            "BAT_PAGER",
            "BAT_CACHE_PATH",
            "BAT_CONFIG_PATH",
            "BAT_OPTS",
            "BAT_STYLE",
            "BAT_TABS",
            "BAT_THEME",
            "XDG_CONFIG_HOME",
            "XDG_CACHE_HOME",
            "COLORTERM",
            "NO_COLOR",
            "MANPAGER",
        ]))
        .info(FileContent::new("Config file", config_file()))
        .info(FileContent::new(
            "Custom assets metadata",
            custom_assets_metadata,
        ))
        .info(DirectoryEntries::new(
            "Custom assets",
            PROJECT_DIRS.cache_dir(),
        ))
        .info(CompileTimeInformation::default());
```

## Filename and Language selected

Rust (`.rs`)

## Versions used

* VSCode version: `1.65.1`
* Theme version: `v2.2.3`

**Comment by wentwrong** [comment]
For anyone stumbling upon the inlay hints coloring problem for js and ts:
1. Open settings.json file (`Ctrl + Shift + P` -> `Preferences: Open Settings (JSON)`)
2. Add the following lines
```json
"javascript.inlayHints.parameterNames.enabled": "all",
"typescript.inlayHints.parameterNames.enabled": "all",
"workbench.colorCustomizations": {
    "[Atom One Dark]": {
        "editorInlayHint.background": "#2c323d",
        "editorInlayHint.foreground": "#9ba1ad"
    }
}
```

**Comment by joeyame** [comment]
@akamud 

This is definitely a big issue and could be fixed with a max of 30 mins of work. I'd be happy to do it, but only if it would eventually get merged. Seeing the two pull requests open for the last two years is worrying though. Is this still maintained?

For others looking for an alternative, just get the One Dark Pro theme. Looks pretty similar and they declare inlayHints correctly.

**Comment by akamud** [comment]
This is now fixed in 2.3.0.
